src/test/java/seedu/address/logic/parser/ViewCommandParserTest.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseSuccess;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_PERSON;
+import static seedu.address.testutil.TypicalIndexes.INDEX_THIRD_PERSON;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.ViewCommand;
+
+public class ViewCommandParserTest {
+
+    private final ViewCommandParser parser = new ViewCommandParser();
+
+    @Test
+    public void parse_validArgs_returnsViewCommand() {
+        assertParseSuccess(parser, "1", new ViewCommand(INDEX_FIRST_PERSON));
+        assertParseSuccess(parser, "3", new ViewCommand(INDEX_THIRD_PERSON));
+    }
+
+    @Test
+    public void parse_validArgsWithWhitespace_returnsViewCommand() {
+        assertParseSuccess(parser, "     1", new ViewCommand(INDEX_FIRST_PERSON));
+        assertParseSuccess(parser, "3    ", new ViewCommand(INDEX_THIRD_PERSON));
+        assertParseSuccess(parser, " \n 2 \t ", new ViewCommand(INDEX_SECOND_PERSON));
```

Comment:
How does this line succeed? Aren't the \n \t misinputs? When I tried it in the actual console it failed.

Reply:
The `\n` and `\t` are special characters here to indicate new line and tab respectively, and not the `n/` `t/` prefixes we use for specific types of inputs.

This test will never come to pass though, since the user cannot enter newline or tab characters within the command input box. It's meant to be a check to ensure that we never lose the functionality to ignore excess whitespace within the `ViewCommandParser`, in case we change the UI in the future and allow users to enter newline or tab characters! 

Hope this helps, do let me know if more clarification is required :)